cpp/include/cudf/types.hpp

```diff
@@ -222,4 +223,9 @@ inline bool operator==(data_type const& lhs, data_type const& rhs) {
  */
 std::size_t size_of(data_type t);
 
+/**
+ * @brief Define concrete dictionary type.
+ */
+struct dictionary32_tag{};
```

Comment:
I'm thinking we should do something like this to avoid hardcoding `int32` everywhere for the index type. This way it is abstracted. 
```suggestion
struct dictionary32_tag{
   using index_type = size_type;
};
```

Reply:
This can then be accessed like:
```
using index_type = dictionary32_tag::index_type;

dictionary32_tag::index_type index;

dictionary_column.indices().begin<dictionary32_tag::index_type>();
```

Reply:
Is there a reason to use `size_type` here over `int32_t`?
I feel `int32_t` is more clear and more correct.
When we support the other 3 integer types, we would want their `index_type` to be the appropriate `int*_t` as well.

Reply:
Yeah, `int32_t` makes more sense instead of `size_type`.